synapseclient/client.py

```diff
@@ -621,6 +621,76 @@ def invalidateAPIKey(self):
         if self._is_logged_in():
             self.restDELETE("/secretKey", endpoint=self.authEndpoint)
 
+    @functools.lru_cache()
+    def get_user_name_profile(
```

Comment:
Thinking about naming of these functions:
`def get_user_profile_by_....`
`def get_user_profile_by_username`

might be a bit more consistent - I know when I'm searching for 'like' functions I usually expect the prefix to match.

Reply:
I agree. I'll do something like that